File: components/text.lua
local component = require('library.component')

return function(content, align, padding, colour)
    local c = component('text')
    c.content = {colour or {1, 1, 1, 1}, content or ''}
    c.align = align or 'left'
    c.padding = padding or 0
    return c
end

File: component/text.go
package component

import (
	"image/color"

	"golang.org/x/image/font"
)

type TextLine struct {
	Content string
	X       float64
}

type TextComponent struct {
	Lines      []TextLine
	Color      color.NRGBA
	LineHeight int
	Font       font.Face
	IsCentered bool
	Padding    float64
}

func NewTextComponent(text string) *TextComponent {
	c := &TextComponent{}
	c.Lines = append(c.Lines, TextLine{Content: text})
	return c
}
